hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -14,8 +14,8 @@ const (
 	EC2VolumeDefaultType string = "gp3"
 )
 
-func machineDeployment(nodePool *hyperv1.NodePool, clusterName string, controlPlaneNamespace string) *capiv1.MachineDeployment {
-	resourcesName := generateName(clusterName, nodePool.Spec.ClusterName, nodePool.GetName())
+func machineDeployment(nodePool *hyperv1.NodePool, controlPlaneNamespace string) *capiv1.MachineDeployment {
+	resourcesName := generateName("", nodePool.Spec.ClusterName, nodePool.GetName())
```

Comment:
I think we should probably just use nodePool.GetName() here as in machineSet(), then follow up with some generator to possibly include the ns to avoid potential cloud provider naming clashing. That's orthogonal to this PR so feel free to ignore anyway.